File: gang/week3/가까운 수.java
class Solution {
    public int solution(int[] array, int n) {
        int answer = array[0];  // 배열의 첫번째 값을 초기값으로 지정

        for (int i = 1; i < array.length; i++) {
            int dist1 = Math.abs(answer - n);   //Math.abs함수로 n과의 거리를 절대값으로 구함
            int dist2 = Math.abs(array[i] - n);

            //거리를 구해서 더 가까운쪽, 거리가 같으면 크기가 더 작은쪽을 answer에 대입
            if (dist2 < dist1 || (dist1 == dist2 && array[i] < answer)) {
                answer = array[i];
            }
        }
        return answer;
    }
}

File: jeonghyeon/3week/가까운 수.js
function solution(array, n) {
    const differ = []; // 차이 배열 만들기
    for (let i = 0; i < array.length; i++) { 
        differ.push(Math.abs(array[i] - n));
    } // 각 요소랑 n과의 차이 계산해 배열에 추가
    const mindiffer = Math.min(...differ);
    // differ 배열에서 가장 작은 값 찾기
    
    let minnum = null; // minnum 변수를 초기값으로 설정
    for (let i = 0; i < array.length; i++) {
        if (differ[i] == mindiffer) { // 가장 작은 값과 같을 때
            if (minnum == null || array[i] < minnum) { // 더 작으면 바꿈
                minnum = array[i];
            }
        }
    }
    return minnum; // 가장 가까운 수 반환
}